### ISSUE TITLE
Document the shift in the criteria for inclusion of rules

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8159]
### Describe the documentation issue

In two places (so far), we propose adding new (and more powerful) rules that overlap with existing ones:

- https://github.com/stylelint/stylelint/issues/8133
- https://github.com/stylelint/stylelint/issues/7557

If we go down this route, we should update our documentation to reflect it.

### What solution would you like to see?

We can:

- change "singular purpose that doesn't overlap with other rules" to "singular purpose" in our [writing rules doc](https://stylelint.io/developer-guide/rules)
- add a corresponding "Configurable" section to match our "configurable - options to tailor the linter" bullet in [our vision doc](https://stylelint.io/about/vision) where we can describe offering a choice of rules that sometimes cover the same area

Going forward, we can reference (and highlight the differences of) overlapping rules in the extended descriptions of the appropriate rule READMEs.

**Comment by Mouvedia** [comment]
> change "singular purpose that doesn't overlap with other rules" to "singular purpose" in our [writing rules doc](https://stylelint.io/developer-guide/rules)

Are there combinations that would results in undefined behaviours?
i.e. incompatible rules' configurations
If so:
- can you can you give an example?
- should we detect and report them?

**Comment by jeddy3** [comment]
Since the beginning we've had overlapping `(dis)allowed-list` rules:

```json
{
  "rules": {
    "unit-disallowed-list": ["px"],
    "unit-allowed-list": ["px"]   
  }
}
```

```css
a {
  margin: 1px;
}
```

```console 
2:11-13 error Unexpected unit "px" (unit-disallowed-list)
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hQEYAHADwG4sBfEAGhADNSAbGAOSR44iGHSE0+AOjABnWe3ARM3YghAZMKFOhCEArn1m7Km7dt37MpHAFoopWUh48IAdxhRbPRzhMoAbV16XQBdNixzHRArG1tnVw8vH1k-BEDgujDzSJRmFkVIFVJiADEIQgIcdQArWWVFWBoFRDNo1IBPPh9MNMpdHiQcGFTdCK12nC6YHrsi1VtUpEwoIih-AaGRtILmIA).

More recently, we have overlapping rules that flag problems twice:

```json
{
  "rules": {
    "function-no-unknown": true,
    "declaration-property-value-no-unknown": true
  }
}
```

```css
a {
  margin: unknown();
}
```

```console
2:10-17 error Unexpected unknown function "unknown" (function-no-unknown)
2:10-19 error Unexpected unknown value "unknown()" for property "margin" (declaration-property-value-no-unknown)
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgWyQCcBzAS03hQFdMBrTCAd0wAoBKAbiwF8QAaEADNSAGxgA5JHjiIYADykAHMQDowAZ3X9wETMOIIQGTChToQhKmPXnKx06fOCaYHKV0BaBh5r0m2BBQcSxg+LAczEFgwESIkN09FQghFGEIcAE8PADckESoYLwgfOgZmWyCQ8JRuHm1IPVJiADEIQgIcQwArdV1tWEUtRHtI9UyxEXIcCvNYnBgx8zCTUfGYScwcDwb9DzGkTCgiKBmQOYXpkDruIA).

We've just let users resolve the conflicts in their config. No one has raised it as a problem so far. 

The new rules I linked to in https://github.com/stylelint/stylelint/issues/8159#issue-2707477117 are more of the latter. We'll turn most of them on in our recommended config and resolve overlaps on the user's behalf.

**Comment by ybiquitous** [comment]
This suggestion sounds great to me 👍🏼
